extension/dynamicconfig/model/config.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"strings"
+
+	com "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1"
+	res "github.com/open-telemetry/opentelemetry-proto/gen/go/resource/v1"
+)
+
+type Config struct {
+	ConfigBlocks []*ConfigBlock // TODO: ok with viper? then convert others, too
+}
+
+func (config *Config) Match(resource *res.Resource) *ConfigBlock {
+	resourceSet, resourceList := embed(resource)
+	totalBlock := &ConfigBlock{
+		Resource:     resourceList,
+		MetricConfig: &MetricConfig{},
+	}
+
+	for _, block := range config.ConfigBlocks {
+		if doInclude(block, resourceSet) {
+			totalBlock.Add(block)
+		}
+	}
+
+	return totalBlock
+}
+
+func embed(resource *res.Resource) (map[string]bool, []string) {
+	resourceSet := make(map[string]bool)
```

Comment:
should this be called labelSet and labelList instead?

It is a set of labels instead of resources, right?

Reply:
Yup! Great point, renaming